src/field/MagField.hh

```diff
@@ -0,0 +1,47 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file MagField.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Array.hh"
+#include "base/Macros.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The MagField evaluates the magnetic field value at a given position.
+ */
+class MagField
+{
+  public:
+    //@{
+    //! Type aliases
+    using field_value = Real3;
+    using field_type  = bool;
```

Comment:
I think these are unnecessary. Change `MagField` to be `UniformMagField`, and change `operator()` to take an `OdeState` (or `FieldTrackView` if you change it to be more state-like) so that the same function-like interface can be used if you extend the field state to include time dependence.

When we extend the field to be nonuniform, we'll change the class that's currently called `FieldEquation` so that it's instantiated on this field class. At the end of the day I think we should have one field propagation kernel per magnetic field parameterization -- one for uniform, one for meshed, etc.

You can delete the `get_field` accessor once you make this change.

Reply:
Removed the type aliases.  Actually UniformMagField may not be a good choice as we can always set the value as actual value (no need a class for that).  The intention for having MagField is to make a general interface for supporting user defined magnetic field implementation (whether it is a map or functional parametrization). A uniform magnetic field value here is a temporary artifact for testing and validating the stepper for development.  In reality, there is no uniform magnetic field used in real experiments and no need of the (RK-type of) stepper as the solution is simply a helix trajectory.  However, I admit that the MagField interface may be completely move the equation, but thought the current approach may give more flexibility for cases with non-magnetic field (which I will introduce later).

Reply:
Makes sense. We can work out the multiple "Field" use cases and different magnetic field types later.